gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -46,11 +82,50 @@
 
     public static final String REFASTER_CONFIGURATION = "refaster";
     public static final String EXTENSION_NAME = "baselineErrorProne";
+    public static final String PROP_STRICT = "com.palantir.baseline-error-prone.strict";
 
     private static final String ERROR_PRONE_JAVAC_VERSION = "9+181-r4173-1";
     private static final String PROP_ERROR_PRONE_APPLY = "errorProneApply";
     private static final String PROP_REFASTER_APPLY = "refasterApply";
 
+    private static final ImmutableSet<BugCheckerInfo> allBaselineChecks =
+            BuiltInCheckerSuppliers.getSuppliers(ImmutableList.of(
```

Comment:
error-prone maintains collections of their `ENABLED_ERRORS `, `ENABLED_WARNINGS`, `DISABLED_CHECKS`, see https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/scanner/BuiltInCheckerSuppliers.java

Thoughts on exposing the baseline versions of these to consumers via `public static` method?

Reply:
That would be helpful, I know of a large internal project which would benefit from the list.

Reply:
Can repos still disable certain checks? In particular I'm worried about `SwitchStatementDefaultCase` as there are probably some valid uses of default cases, most notably with non-API enum with many variants.

Reply:
You can `@SuppressWarnings("SwitchStatementDefaultCase")` for targeted classes or methods today.

We could potentially allow suppressing specific checks as part of baseline gradle config, but defer to baseline infra folks if that’s something we want to allow.

Reply:
I'm referring to modifying the error-prone options, which you can do today and which I assume will still work after this change.

Reply:
^^^ would be good test cases, I don’t know offhand how the ordering in current PR would handle project adjusted error prone check args.

Reply:
@pkoenig10 added the ability to explicitly disable a specific check in strict mode (open to feedback on structure (as is this just leaves the specific check at its default severity level instead of promoting it to error -- do we actually want more control over severity?)

```
baselineErrorProne {
    strict true
    disableCheck 'SwitchStatementDefaultCase'
}
```

One can still use something along the lines of `options.errorprone.errorproneArgs << '-Xep:NullAway:ERROR'` as [`net.ltgt.gradle.errorprone.ErrorProneOptions#toString()`](https://github.com/tbroyer/gradle-errorprone-plugin/blob/master/src/main/kotlin/net/ltgt/gradle/errorprone/ErrorProneOptions.kt#L66-L84) is used to output all the compiler args and those will be output after the `checks` per the following sequence :

```
        return (
            sequenceOf(
                "-XepDisableAllChecks".takeIf { disableAllChecks },
                "-XepAllErrorsAsWarnings".takeIf { allErrorsAsWarnings },
                "-XepAllDisabledChecksAsWarnings".takeIf { allDisabledChecksAsWarnings },
                "-XepDisableWarningsInGeneratedCode".takeIf { disableWarningsInGeneratedCode },
                "-XepIgnoreUnknownCheckNames".takeIf { ignoreUnknownCheckNames },
                "-XepCompilingTestOnlyCode".takeIf { isCompilingTestOnlyCode },
                "-XepExcludedPaths:$excludedPaths".takeUnless { excludedPaths.isNullOrEmpty() }
            ).filterNotNull() +
                checks.asSequence().onEach(::validateName).map { (name, severity) -> "-Xep:$name${severity.asArg}" } +
                checkOptions.asSequence().map { (name, value) -> "-XepOpt:$name=$value" } +
                errorproneArgs +
                errorproneArgumentProviders.asSequence().flatMap { it.asArguments().asSequence() }
            ).onEach(::validate)
            .joinToString(separator = " ")

```

Reply:
Do we actually need this `disableCheck`? Is it not sufficient to do:
```
    tasks.withType(JavaCompile) {
        options.errorprone.errorproneArgs += ['-Xep:SwitchStatementDefaultCase:OFF']
    }
```